tests/common/include/tests/common/path_correction.inl

```diff
@@ -7,463 +7,171 @@
+
+    // Copy the covariance
+    typename bound_track_parameters<transform3>::covariance_type bound_cov =
+        bound_cov0;
+
+    // Total path length, just for testing purpose
+    scalar total_path_length = 0.f;
+
+    // Iterate over the planes until we reach the first plane (one loop)
+    for (std::size_t i_p = 0u; i_p < n_planes; i_p++) {
+
+        // Next surface index (circular)
+        const std::size_t next_index = i_p == n_planes - 1 ? 0u : i_p + 1;
+
+        // Helix on the current surface
+        detail::helix<transform3> hlx(free_trk, &B);
+
+        // bound vector on the current surface
+        const auto bound_vec =
+            c2.free_to_bound_vector(trfs[i_p], free_trk.vector());
+
+        // Bound-to-free jacobian on the current surface
+        const typename cartesian2<transform3>::bound_to_free_matrix
+            bound_to_free_jacobi =
+                c2.bound_to_free_jacobian(trfs[i_p], rectangle, bound_vec);
+
+        // Get the intersection on the next surface
+        const auto is = hpi(hlx, rectangle, trfs[next_index])[0];
+
+        // Helical path length between two surfaces
+        const auto path_length = is.path;
+
+        // Position on the next surface
+        const auto new_pos = hlx(path_length);
+
+        // Direction on the next surface
+        const auto new_dir = hlx.dir(path_length);
+
+        // Add the path length to the total path length
+        total_path_length += path_length;
+
+        // Free transport jacobian between two surfaces
+        const cartesian2<transform3>::free_matrix transport_jacobi =
+            hlx.jacobian(path_length);
+
+        // Reset the free track parameters with the position and direction on
+        // the next surface
+        free_trk.set_pos(new_pos);
+        free_trk.set_dir(new_dir);
+
+        // Path correction
+        const cartesian2<transform3>::free_matrix path_correction =
+            c2.path_correction(free_trk, B, trfs[next_index], rectangle, true);
+
+        // Correction term for the path variation
+        const cartesian2<transform3>::free_matrix correction_term =
+            matrix_operator().template identity<e_free_size, e_free_size>() +
+            path_correction;
+
+        // Free to bound jacobian
+        const typename cartesian2<transform3>::free_to_bound_matrix
+            free_to_bound_jacobi = c2.free_to_bound_jacobian(
+                trfs[next_index], rectangle, free_trk.vector());
+
+        // Full jacobian
+        const typename cartesian2<transform3>::bound_matrix full_jacobi =
+            free_to_bound_jacobi * correction_term * transport_jacobi *
+            bound_to_free_jacobi;
+
+        // Update the covariance at the next surface
+        bound_cov =
+            full_jacobi * bound_cov * matrix_operator().transpose(full_jacobi);
     }
 
-    // covariance
+    // Check if the total path length is the expected value
+    ASSERT_NEAR(total_path_length, S, tolerance);
```

Comment:
Hm, once we have the ```path_correction``` term, do we really need to calculate the ```bound_cov``` from it? This is not the same covariance transport code that we use in the propagation, so how much do we actually gain by testing it? Is there really no other way to see that the ```path_correction``` term makes sense? While it is nice that the ```navigator```, ```propagator``` etc. is gone now, this still feels more like an integration test than a unittest

Reply:
> Is there really no other way to see that the path_correction term makes sense?

I would like to know if there is any other way to test it. That was really my best yeah but you are right - it looks like integration test rather than unit test.

Reply:
I think I will change to parameter_transport.inl or sth like that

Reply:
The name changed to `covariance_transport.inl`